clang/lib/AST/Expr.cpp

```diff
@@ -4806,6 +4806,34 @@ getOverflowPatternBinOp(const BinaryOperator *E) {
   return {};
 }
 
+/// Compute and set the OverflowPatternExclusion bit based on whether the
```

Comment:
Maybe this way?
```
static void computeOverflowPatternExclusion(const ASTContext &Ctx,
                                            const BinaryOperator *E) {
  std::optional<BinaryOperator *> Result = getOverflowPatternBinOp(E);
  if (!Result.has_value())
    return;
  QualType AdditionResultType = Result.value()->getType();

  bool AddSignedOverflowTest = Ctx.getLangOpts().isOverflowPatternExcluded(
      LangOptions::OverflowPatternExclusionKind::AddSignedOverflowTest) &&
     AdditionResultType->isSignedIntegerType();
  bool AddUnsignedOverflowTest = Ctx.getLangOpts().isOverflowPatternExcluded(
      LangOptions::OverflowPatternExclusionKind::AddUnsignedOverflowTest)
      && AdditionResultType->isUnsignedIntegerType();

  if (AddSignedOverflowTest || AddUnsignedOverflowTest)
     Result.value()->setExcludedOverflowPattern(true);
}
```

Reply:
I wanted to avoid calling `getOverflowPatternBinOp()` as it is more expensive than just checking if the flags are enabled or not.